src/grafana_agent.py

```diff
@@ -387,6 +395,23 @@ def _update_status(self, *_):
 
         self.unit.status = ActiveStatus()
 
+    def _has_outgoing(self, outgoings) -> bool:
+        for outgoing in outgoings:
+            relation = self.model.relations.get(outgoing, [])
+
+            try:
+                units = relation[0].units
+            except IndexError:
+                units = None
+
+            if relation == []:
+                continue
```

Comment:
**nitpick**: these two blocks could be swapped because you already have the relation object, so you could enter the `continue` without checking for the unit

```suggestion
            if relation == []:
                continue

            try:
                units = relation[0].units
            except IndexError:
                units = None
```